lib/agenda/database.js

```diff
@@ -1,42 +1,34 @@
-'use strict';
-const MongoClient = require('mongodb').MongoClient;
+const {promisify} = require('util');
+const {MongoClient} = require('mongodb');
 const debug = require('debug')('agenda:database');
 
 /**
- * Connect to the spec'd MongoDB server and database.
+ * Connect to the MongoDB server and database.
  * @param {String} url MongoDB server URI
  * @param {String} collection name of collection to use
  * @param {Object} options options for connecting
- * @param {Function} cb callback of MongoDB connection
- * @returns {exports}
+ * @returns {Promise}
  * NOTE:
  * If `url` includes auth details then `options` must specify: { 'uri_decode_auth': true }. This does Auth on
  * the specified database, not the Admin database. If you are using Auth on the Admin DB and not on the Agenda DB,
  * then you need to authenticate against the Admin DB and then pass the MongoDB instance into the constructor
  * or use Agenda.mongo(). If your app already has a MongoDB connection then use that. ie. specify config.mongo in
  * the constructor or use Agenda.mongo().
  */
-module.exports = function(url, collection, options, cb) {
-  const self = this;
-  if (!url.match(/^mongodb:\/\/.*/)) {
-    url = 'mongodb://' + url;
-  }
+module.exports = function(url, collection = 'agendaJobs', options) {
+  return new Promise(async (resolve, reject) => {
+    options = Object.assign({autoReconnect: true, reconnectTries: Number.MAX_SAFE_INTEGER, reconnectInterval: this.config.processEvery}, options);
+    if (!url.match(/^mongodb:\/\/.*/)) {
+      url = 'mongodb://' + url;
+    }
 
-  collection = collection || 'agendaJobs';
-  options = Object.assign({autoReconnect: true, reconnectTries: Number.MAX_SAFE_INTEGER, reconnectInterval: this._processEvery}, options);
-  MongoClient.connect(url, options, (error, db) => {
-    if (error) {
+    // @TODO: This should be the mongodb instance
+    this.ctx.mongo = await promisify(MongoClient.connect)(url, options).catch(err => {
```

Comment:
Anyone know why this would be returning undefined instead of the mongodb object?